data-runtime/src/main/java/io/micronaut/data/runtime/mapper/sql/SqlResultEntityTypeMapper.java

```diff
@@ -668,7 +668,10 @@ private <K> K buildIdOnlyEntity(RS rs, MappingContext<K> ctx, Object resolvedId)
                         if (resolvedId == null) {
                             resolvedId = readEntityId(rs, ctx);
                         }
-                        return associatedIntrospection.instantiate(resultReader.convertRequired(resolvedId, identity.getType()));
+                        // instantiate only if id value is resolved
+                        if (resolvedId != null) {
```

Comment:
This happens when many to one entity has only id and fk value is not set (null). So just adding check solves this minor bug I noticed.